File: backend/src/main/java/newstock/domain/news/entity/News.java
package newstock.domain.news.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="news")
public class News {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer newsId;

    private Integer stockId;

    private String title;

    private String description;

    @Lob
    private String content;

    private String newsImage;

    private String url;

    private String press;

    private String pressLogo;

    private String publishedDate;

    @Lob
    private String newsSummary;

    private int score;

}

File: frontend/types/api/news.ts
import { BaseResType } from './base';

export type AllStockNewsResType = BaseResType & {
  data: {
    totalPage: number;
    newsList: {
      newsId: number;
      title: string;
      score: number;
      publishedDate: string;
    }[];
  };
};

export type topFiveStockNewsResType = BaseResType & {
  data: {
    newsList: {
      newsId: number;
      title: string;
      score: number;
      publishedDate: string;
    }[];
  };
};

export type NewsScrapResType = BaseResType & {
  data: {
    totalPage: number;
    newsList: NewsType[];
  };
};

export type NewsType = {
  newsId: number;
  title: string;
  description: string;
  score: number;
  publishedDate: string;
};

export type NewsDetailResType = BaseResType & {
  data: {
    newsInfo: {
      title: string;
      content: string;
      newsImage: string;
      url: string;
      press: string;
      pressLogo: string;
      publishedDate: string;
      newsSummary: string;
      score: number;
    };
    isScraped: boolean;
  };
};
